### DISCUSSION TITLE
FPrime Topology visualisation using Diagrams.Net

### DISCUSSION THREAD

**Discussion by IonRod** [nasa/fprime#868] [Show and tell]
Hi all!

We've made a simple [Python script](https://github.com/spacebit-apps/fprime-top2diagrams.net) that visualises topology using Diagrams.net.

At this point in time it's an internal script, so it would be great to hear if this script could be useful for you. It would endorse us to improve the script.

Here is a screenshot of the topology of the Ref application.

![Ref](https://user-images.githubusercontent.com/886012/126784197-e6dfef0d-102b-4101-9074-db2715915970.jpg)

The layout is far from being perfect, but it's interactive, so you can move and organise nodes manually by hand.

Here is the link to generated drawio file. (It should be unzipped).
[Ref.drawio](https://github.com/spacebit-apps/fprime-top2diagrams.net/blob/main/Output/Ref.drawio)

To view the file desktop Diagrams.Net (former Draw.io) app should be downloaded: [Diagrams.Net Desktop app](https://github.com/jgraph/drawio-desktop/releases)

**Comment by knapsea** [comment]
with provided Ref.drawio file , fails as below at  load , this is with latest binary version 14.6.13

Not a diagram file (error on line 90 at column 90: Specification mandates value for attribute data-pjax-transient)

**Reply by IonRod** [reply]
Hi, thank you for reply!

That's a weird thing. 

I've downloaded the file and tested on Mac, Windows and using [web version of Diagrams.net](https://app.diagrams.net/)

Could you try to:
1. Upload the downloaded file here.
2. Download the file ones again. (Maybe something happened during download?)
2. Try with [web version of Diagrams.net](https://app.diagrams.net/)

Thank you in advance!

**Comment by knapsea** [comment]
Using  https://app.diagrams.net/

Same error with web and 2nd time downloaded file.
What was is about " (It should be unzipped)" , I get a  .drawio file not a .zip file

**Reply by spacebit-official** [reply]
Hi!

Could you send your OS and the file that you've downloaded?

It looks like the file wasn't downloaded complitely.
